File: Forth_Interpreter/Dup.cpp
#include "Dup.hpp"
#include "exceptions.hpp"

#include <memory>

bool Dup::Run () {
    if (GeneralDataForExecutors.Stack->empty()){
        throw Exception_EmptyStack();
    }

    GeneralDataForExecutors.Stack->push (GeneralDataForExecutors.Stack->top());

    return false;
}

File: Forth_Interpreter_On_Java/Dup/Dup.java
package Dup;

import ForthCommands.ForthCommands;
import java.util.EmptyStackException;

public class Dup extends ForthCommands{
    
    @Override
    public boolean Run() {
        try {
            GeneralDataForExecutors.stack.push (GeneralDataForExecutors.stack.peek());
        } catch (EmptyStackException e) {
            System.err.println(e.getMessage());
        }

        return false;
    }
}
